src/couch_mrview_http.erl

```diff
@@ -136,7 +141,7 @@ handle_view_req(Req, _Db, _DDoc) ->
 
 
 handle_temp_view_req(#httpd{method='POST'}=Req, Db) ->
-    couch_httpd:validate_ctype(Req, "application/json"),
+    chttpd:validate_ctype(Req, "application/json"),
```

Comment:
Here you first check content type and the if user is admin, while below,  handle_compact_req and handle_cleanup_req does that in reverse. Let's keep some single flow.